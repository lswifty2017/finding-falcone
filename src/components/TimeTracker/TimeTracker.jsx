import React from 'react';
import './TimeTracker.scss';

const TimeTracker = ({ planets = [], vehicles = [] }) => {
  let totalTime = 0;

  planets.forEach((planet) => {
    vehicles.forEach((vehicle) => {
      if (vehicle.selected.includes(planet.selected)) {
        totalTime += planet.distance / vehicle.speed;
      }
    });
  });

  return (
    <div className="time-tracker">
      Total Time Taken: <span id="total-time">{totalTime}</span>
    </div>
  );
};

export default TimeTracker;
