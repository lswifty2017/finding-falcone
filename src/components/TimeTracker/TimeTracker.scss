.time-tracker {
  font-size: 20px;
  padding-bottom: 24px;
}
