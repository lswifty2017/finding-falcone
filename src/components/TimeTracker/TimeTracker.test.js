import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import TimeTracker from './TimeTracker.jsx';

describe('TimeTracker', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TimeTracker />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calculates total time for one vehicle selected for one planet', () => {
    const mockData = {
      planets: [{ name: 'Donlon', selected: 'card-1', distance: 50 }],
      vehicles: [{ name: 'Space Pod', selected: ['card-1'], speed: 2 }],
    };

    const expectedTime =
      mockData.planets[0].distance / mockData.vehicles[0].speed;

    const { container } = render(<TimeTracker {...mockData} />);

    const timeElement = container.querySelector('#total-time');

    expect(parseInt(timeElement.innerHTML)).toEqual(expectedTime);
  });

  it('calculates total time for multiple vehicles and planets', () => {
    const mockData = {
      planets: [
        { name: 'Donlon', selected: 'card-1', distance: 50 },
        { name: 'Jebing', selected: 'card-2', distance: 100 },
      ],
      vehicles: [
        { name: 'Space Pod', selected: ['card-1'], speed: 2 },
        { name: 'Space Rocket', selected: ['card-2'], speed: 1 },
      ],
    };

    const expectedTime =
      mockData.planets[0].distance / mockData.vehicles[0].speed +
      mockData.planets[1].distance / mockData.vehicles[1].speed;

    const { container } = render(<TimeTracker {...mockData} />);

    const timeElement = container.querySelector('#total-time');

    expect(parseInt(timeElement.innerHTML)).toEqual(expectedTime);
  });

  it('calculates total time for multiple vehicles for 1 planets', () => {
    const mockData = {
      planets: [
        { name: 'Donlon', selected: 'card-1', distance: 50 },
        { name: 'Jebing', selected: 'card-2', distance: 100 },
      ],
      vehicles: [
        { name: 'Space Rocket', selected: ['card-2', 'card-1'], speed: 1 },
      ],
    };

    const expectedTime =
      mockData.planets[0].distance / mockData.vehicles[0].speed +
      mockData.planets[1].distance / mockData.vehicles[0].speed;

    const { container } = render(<TimeTracker {...mockData} />);

    const timeElement = container.querySelector('#total-time');

    expect(parseInt(timeElement.innerHTML)).toEqual(expectedTime);
  });

  it('calculates total time to be 0 if planet and vehicle are not selected', () => {
    const mockData = {
      planets: [{ name: 'Donlon', selected: false, distance: 50 }],
      vehicles: [{ name: 'Space Rocket', selected: [], speed: 1 }],
    };

    const expectedTime = 0;

    const { container } = render(<TimeTracker {...mockData} />);

    const timeElement = container.querySelector('#total-time');

    expect(parseInt(timeElement.innerHTML)).toEqual(expectedTime);
  });
});
