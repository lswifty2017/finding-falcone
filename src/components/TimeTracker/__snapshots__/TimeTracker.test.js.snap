// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeTracker renders correctly 1`] = `
<div
  className="time-tracker"
>
  Total Time Taken: 
  <span
    id="total-time"
  >
    0
  </span>
</div>
`;
